doc/man3/OSSL_PARAM_allocate_from_text.pod

```diff
@@ -28,28 +28,21 @@ size (see L<OSSL_PARAM(3)> for more information).
 
 OSSL_PARAM_allocate_from_text() takes a control I<key>, I<value> and
 value size I<value_n>, and given a parameter descriptor array
-I<paramdefs>, it converts the value to something suitable for
-L<OSSL_PARAM(3)> and stores that in the buffer I<buf>, and modifies
-the parameter I<to> to match.
-I<buf_n>, if not NULL, will be assigned the number of bytes used in
-I<buf>.
-If I<buf> is NULL, only I<buf_n> will be modified, everything else is
-left untouched, allowing a caller to find out how large the buffer
-should be.
-I<buf> needs to be correctly aligned for the type of the B<OSSL_PARAM>
-I<key>.
-If <found> is not NULL, it is set to 1 if the parameter can be located and
-to 0 otherwise.
-
-The caller must remember to free the data of I<to> when it's not
-useful any more.
+I<paramdefs>, it converts the value to something suitable for an
+L<OSSL_PARAM(3)> item, and stores the result in I<to>.
+
+The I<data> field in I<to> allocated using L<OPENSSL_zalloc(3)> and
+needs to be freed by the caller when it's not useful any omre, using
+L<OPENSSL_free(3)>.
+
+If I<found> is not NULL, I<*found> is set to 1 if the parameter could
+be located in I<paramdefs>, and to 0 otherwise.
 
 For parameters having the type B<OSSL_PARAM_INTEGER>,
-B<OSSL_PARAM_UNSIGNED_INTEGER>, or B<OSSL_PARAM_OCTET_STRING>, both
-functions will interpret the I<value> differently if the key starts
-with "hex".
-In that case, the value is decoded first, and the result will be used
-as parameter value.
+B<OSSL_PARAM_UNSIGNED_INTEGER>, or B<OSSL_PARAM_OCTET_STRING>,
+OSSL_PARAM_allocate_from_text() will interpret the I<value>
+differently if the key starts with "hex".  In that case, the value is
+hex decoded first, and the result will be used as parameter value.
```

Comment:
I think this at least needs an example. I actually assume it start with "hex:", as in "hex:4F", and not "hex4F".

Reply:
Ah, thank you for pointing that out.

Well, yes and no. It's rather that this corresponds to things like `-pkeyopt id:A` vs `-pkeyopt hexid:41`. The key *starts with* "hex", and with OSSL_PARAM_allocate_from_text(), both forms will become an OSSL_PARAM item with the key "id" and exactly the same value.

Reply:
I've added a little more text that I hope clarifies this